spec/service/default_group_service_spec.rb

```diff
@@ -0,0 +1,121 @@
+require "rails_helper"
+
+RSpec.describe DefaultGroupService do
+  subject(:default_group_service) do
+    described_class.new
+  end
+
+  describe "#create_trial_user_default_group!" do
+    let(:user) { create :user, name: "Batman" }
+    let(:form) { build :form, id: 1 }
+    let(:forms_response) do
+      [form]
+    end
+
+    before do
+      ActiveResource::HttpMock.respond_to do |mock|
+        mock.get "/api/v1/forms?creator_id=#{user.id}", headers, forms_response.to_json, 200
+      end
+    end
+
+    context "when user has a trial role" do
+      it "creates a group if the user does not already have one" do
+        expect {
+          default_group_service.create_trial_user_default_group!(user)
+        }.to change(Group, :count).by(1)
+
+        expect(Group.last).to have_attributes(
+          creator: user,
+          name: "Batman’s trial group",
+          organisation: user.organisation,
+        )
+        expect(Group.last).to be_trial
+        expect(Group.last.users).to eq [user]
+        expect(Group.last.users.group_admins).to eq [user]
+      end
+
+      it "does not create a group if the user already has one" do
+        group = create :group, creator: user, organisation: user.organisation, name: "Batman’s trial group", status: :trial
+
+        expect {
+          default_group_service.create_trial_user_default_group!(user)
+        }.not_to change(Group, :count)
+        expect(group).to be_trial
+        expect(group.users).to include user
+        expect(Group.last.users.group_admins).to include user
+      end
+
+      it "adds their trial forms to the group" do
+        expect {
+          default_group_service.create_trial_user_default_group!(user)
+        }.to change(GroupForm, :count).by(1)
+
+        expect(GroupForm.last).to have_attributes(form_id: form.id)
+      end
+
+      it "raises an exception if it tries to add their trial forms to an active group" do
```

Comment:
This could be clearer about what scenario this can occur under. Something like:

```suggestion
      when "a default group already exists with active status" do
          when "the user has forms without a group" do
              it "raises an exception" do
```

Reply:
I've rearranged the contexts in a way I _think_ is more logical. See what you think!